File: BE/Src/MIT.Fwk.Infrastructure/Entities/GeoMapping.cs
using JsonApiDotNetCore.Resources;
using JsonApiDotNetCore.Resources.Annotations;

namespace MIT.Fwk.Infrastructure.Entities
{
    [Resource]
    public class GeoMapping : Identifiable<int>
    {
        [Attr]
        public string ZipCode { get; set; }
        [Attr]
        public int GeoFirstDivisionId { get; set; }
        [Attr]
        public int GeoSecondDivisionId { get; set; }
        [Attr]
        public int GeoThirdDivisionId { get; set; }
        [Attr]
        public int GeoCityId { get; set; }
        [Attr]
        public int GeoCountryId { get; set; }
        [HasOne]
        public virtual GeoFirstDivision GeoFirstDivision { get; set; }
        [HasOne]
        public virtual GeoSecondDivision GeoSecondDivision { get; set; }
        [HasOne]
        public virtual GeoThirdDivision GeoThirdDivision { get; set; }
        [HasOne]
        public virtual GeoCity GeoCity { get; set; }
        [HasOne]
        public virtual GeoCountry GeoCountry { get; set; }
    }
}

File: FE/WEB/app/models/geo-mapping.js
import Model, { attr, belongsTo } from '@ember-data/model';

export default class GeoMappingModel extends Model {
  @belongsTo('geo-country', { async: true, inverse: 'geoMappings' })
  geoCountry;
  @belongsTo('geo-city', { async: true, inverse: 'geoMappings' })
  GeoCity;
  @belongsTo('geo-first-division', { async: true, inverse: 'geoMappings' })
  GeoFirstDivision;
  @belongsTo('geo-second-division', { async: true, inverse: 'geoMappings' })
  GeoSecondDivision;
  @belongsTo('geo-third-division', { async: true, inverse: 'geoMappings' })
  GeoThirdDivision;

  @attr('string', { defaultValue: '' }) zipCode;
  @attr('number', { defaultValue: 0 }) geoFirstDivisionId;
  @attr('number', { defaultValue: 0 }) geoSecondDivisionId;
  @attr('number', { defaultValue: 0 }) geoThirdDivisionId;
  @attr('number', { defaultValue: 0 }) geoCityId;
  @attr('number', { defaultValue: 0 }) geoCountryId;
}
